src/main/java/randoop/condition/SpecificationCollection.java

```diff
@@ -0,0 +1,312 @@
+  private Map<AccessibleObject, OperationConditions> conditionMap;
+
+  /**
+   * Creates a {@link SpecificationCollection} for the given specification map.
+   *
+   * <p>This constructor is used internally. It is only accessible to allow testing. Clients should
+   * use {@link #create(List)} instead.
+   *
+   * @param specificationMap the map from reflection objects to {@link OperationSpecification}
+   * @param signatureMap the multimap from a signature to methods with with the signature
+   * @param parentMap the map from a method to methods that it it overrides and that have a
+   *     specification
+   */
+  SpecificationCollection(
+      Map<AccessibleObject, OperationSpecification> specificationMap,
+      MultiMap<OperationSignature, Method> signatureMap,
+      Map<AccessibleObject, Set<Method>> parentMap) {
+    this.specificationMap = specificationMap;
+    this.signatureMap = signatureMap;
+    this.parentMap = parentMap;
+    this.conditionMap = new HashMap<>();
+    SequenceClassLoader sequenceClassLoader = new SequenceClassLoader(getClass().getClassLoader());
+    List<String> options = new ArrayList<>();
+    this.compiler = new SequenceCompiler(sequenceClassLoader, options);
+  }
+
+  /**
+   * Creates a {@link SpecificationCollection} from the list of files of serialized specifications.
+   *
+   * @param specificationFiles the files of serialized specifications
+   * @return the {@link SpecificationCollection} built from the serialized {@link
+   *     OperationSpecification} objects.
+   */
+  public static SpecificationCollection create(List<File> specificationFiles) {
+    if (specificationFiles == null) {
+      return null;
+    }
+    MultiMap<OperationSignature, Method> signatureMap = new MultiMap<>();
+    Map<AccessibleObject, OperationSpecification> specificationMap = new LinkedHashMap<>();
+    for (File specificationFile : specificationFiles) {
+
+      /*
+       * Read the specifications from the file
+       */
+      List<OperationSpecification> specificationList;
+      try {
+        specificationList = readSpecifications(specificationFile);
+      } catch (IOException e) {
+        String msg = "Unable to read specifications from file";
+        if (Log.isLoggingOn()) {
+          Log.logLine(msg + ": " + specificationFile);
+        }
+        throw new RandoopConditionError(msg, e);
+      } catch (Throwable e) {
+        throw new RandoopConditionError("Bad input", e);
+      }
+
+      for (OperationSpecification specification : specificationList) {
+        OperationSignature operation = specification.getOperation();
+
+        // Check for bad input
+        if (operation == null) { // deserialization could result in null operation
+          continue;
+        }
+        if (specification.getIdentifiers().hasDuplicatedName()) {
+          String msg =
+              String.format(
+                  "Ignoring specification with identifier name conflict: %s",
```

Comment:
Since this is illegal user input, I would halt Randoop rather than leaving a user to wonder why a spec was ignored (if, as seems likely, the user didn't notice this in the Randoop output.